content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,290 @@
+    
+We propose a new policy for running Indexed Jobs in which the backoff limit controls the number of retries per index. When the new policy is used all indexes execute until their success or failure. We also propose a new API field to control the number of failed indexes.
+
+Additionally, we propose a new action in PodFailurePolicy, called FailIndex, to short-circuit the failing of the index before the backoff limit per index is reached.
+   
+## cAdvisor-less, CRI-full Container And Pod Stats
+
+This encompasses two related pieces of work (summary API and /metrics/cadvisor), and will require changes in three different components (CRI implementation, Kubelet, cAdvisor).
+
+There are two main APIs that consumers use to gather stats about running containers and pods: summary API and /metrics/cadvisor. The Kubelet is responsible for implementing the summary API, and cadvisor is responsible for fulfilling /metrics/cadvisor.
+
+This aims to enhance CRI implementations to be able to fulfill all the stats needs of Kubernetes. At a high level, there are two pieces of this:
+
+- Enhance the CRI API with enough metrics to be able to supplement the pod and container fields in the summary API directly from CRI.
+    
+- Enhance the CRI implementations to broadcast the required metrics to fulfill the pod and container fields in the /metrics/cadvisor endpoint.
+
+## Feature Graduations And Deprecations In Kubernetes v1.28
+### Graduations to stable
+
+This release includes a total of 12 enhancements promoted to Stable:
+
+* [Kubectl Events](https://github.com/kubernetes/enhancements/issues/1440)
+* [Retroactive default StorageClass assignment](https://github.com/kubernetes/enhancements/issues/3333)
+* [Non-graceful node shutdown](https://github.com/kubernetes/enhancements/issues/2268)
+* [Support 3rd party device monitoring plugins](https://github.com/kubernetes/enhancements/issues/606)
+* [Auth API to get self-user attributes](https://github.com/kubernetes/enhancements/issues/3325)
+* [Proxy Terminating Endpoints](https://github.com/kubernetes/enhancements/issues/1669)
+* [Expanded DNS Configuration](https://github.com/kubernetes/enhancements/issues/2595)
+* [Cleaning up IPTables Chain Ownership](https://github.com/kubernetes/enhancements/issues/3178)
+* [Minimizing iptables-restore input size](https://github.com/kubernetes/enhancements/issues/3453)
+* [Graduate the kubelet pod resources endpoint to GA](https://github.com/kubernetes/enhancements/issues/3743)
+* [Extend podresources API to report allocatable resources](https://github.com/kubernetes/enhancements/issues/2403)
+* [Move EndpointSlice Reconciler into Staging](https://github.com/kubernetes/enhancements/issues/3685)
+
+### Deprecations And Removals
+
+This release saw several removals:
+
+* [Removal of CSI Migration for GCE PD](https://github.com/kubernetes/enhancements/issues/1488)
```

Comment:
Note: there is only single removal based on the current state of the PR, and we might need to reword line 236

Reply:
Thanks @furkatgofurov7 I was wondering about this. What do you recommend I reword it to??

Reply:
Deprecations of Volume Plugin: 

[Ceph RBD in-tree Plugin](https://groups.google.com/g/kubernetes-sig-storage/c/h5751_B5LQM) has been deprecated in v1.28 and is targeted for removal in v1.31 (no CSI migration planned).

[Ceph FS in-tree Plugin](https://groups.google.com/g/kubernetes-sig-storage/c/IlHnIpwTbEg) has been deprecated in v1.28 and is targeted for removal in v1.31 (no CSI migration planned).

PRs:https://github.com/kubernetes/website/pull/42201

Reply:
@bradmccoydev perhaps it can stay as is if we have multiple removals based on @carlory s comments, otherwise I was thinking of something like:

`This release saw only one removal:`

could be a better fit.